### PR TITLE
make rarer artifacts more likely to vary in appearance

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
This makes artifacts with a lower rarity_weight (which appear less often) more likely to have some of the random appearance variations (mostly) based on their origin.

Some of the variations were already so common that I just made the rarity not affect their probability.

Right now the highest tier of artifact will make some things about 5x more likely, so you end up with probabilities like 25%/50%.
The most common artifacts (what used to be tier 1 and 2) will have only slightly higher probabilities.
The numbers are kind of made up, so I am open to suggestions on changing them!
## Why's this needed? 
Ever since the appearance variations are added, I get a little excited when I see a weird looking artifact, thinking, "I wonder what that would do!".
I think it'd be neat if that were actually a small clue that it might be a more special kind of artifact!

### CODE REVIEW

File: code/datums/controllers/artifact_controls.dm [hunk 1]
```diff
@@ -270,7 +275,9 @@ var/datum/artifact_controller/artifact_controls
 
 	post_setup(obj/artifact)
 		. = ..()
-		if(prob(5))
+		var/datum/artifact/AD = artifact.artifact
+		var/rarityMod = AD.get_rarity_modifier()
+		if(prob(5 * rarityMod))
 			artifact.transform = matrix(artifact.transform, rand(-10, 10), MATRIX_ROTATE)
 		if(prob(40))
```
Comment (pali6) [inline code/datums/controllers/artifact_controls.dm:282]: Who not the colour variation too?

**Review by pali6 — CHANGES_REQUESTED** [review]
I don't think the scaling is all that well thought out here. I'd expect the `get_rarity_mod()` proc to return a number in range 0-1 (let's say with 0.1 for lowest tier and 1.0 for highest tier or something like that). In that case doing `prob(10 * rarityMod)` makes it so you set the upper bound of the probability of the effect to 10%. While currently the number is sorta arbitrary.

To compensate this change the base probabilities should likely be increased.

**Comment by pali6** [comment]
Wild thought: To ensure the 0-1 range of the rarity mod it could be exponential. Something like `2 ** (-rarity_weight / some_constant)`.

**Comment by zjdtmkhzt** [comment]
Yeah, using an exponent sounds like a much better idea!

**Review by pali6 — APPROVED** [review]

**Pull request merged**